paygManager: Understand ExpiryTime in terms of CLOCK_BOOTTIME

mwleeds opened the pull request:
The "ExpiryTime" property of the EpgProvider interface is changing to be
in terms of CLOCK_BOOTTIME rather than wall clock time, this commit
changes the paygManager class to account for this.

https://phabricator.endlessm.com/T24300

mwleeds commented:
This shouldn't be merged until after https://github.com/endlessm/eos-payg/pull/28

pwithnall commented on src/shell-util.c:
```diff
@@ -776,3 +776,28 @@ shell_util_get_eos_image_version (void)
+ * the time since boot.
+ *
+ * Returns: the time since some unspecified starting point, in microseconds
+ **/
```
```suggestion
 */
```

pwithnall commented on src/shell-util.c:
```diff
@@ -776,3 +776,28 @@ shell_util_get_eos_image_version (void)
+ * shell_util_get_boottime:
+ *
+ * Like g_get_monotonic_time(), but also includes any time the system is
+ * suspended. Uses CLOCK_BOOTTIME, hence the name, but is not guaranteed to be
```
```suggestion
 * suspended. Uses `CLOCK_BOOTTIME`, hence the name, but is not guaranteed to be
```

pwithnall commented on src/shell-util.c:
```diff
@@ -776,3 +776,28 @@ shell_util_get_eos_image_version (void)
+
+  result = clock_gettime (CLOCK_BOOTTIME, &ts);
+
+  if G_UNLIKELY (result != 0)
```
```suggestion
  if (G_UNLIKELY (result != 0))
```

mwleeds commented:
You're right, it would be more correct to get the clock from the provider and call `get_time()` on it than to encode the assumption that CLOCK_BOOTTIME is used into gnome-shell. But I think that would mean changing the D-Bus interface, so probably not worth it. We probably won't need to make a change like this again.

mwleeds commented:
> Looks good to me, although there’s a remaining GLib.get_real_time() call in js/ui/paygUnlockDialog.js. I think that needs to be replaced with EpgClock usage if the call here is also changed. Although probably every call to g_get_real_time() outside the EpgClock implementation should be changed.

I addressed this with a commit

mwleeds commented:
Merging this now that https://github.com/endlessm/eos-payg/pull/28 has been merged

mwleeds commented:
Oops, forgot to squash the fixup. Sorry
